WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Carriers and Rates/ShippingLabelCarrierRow.swift

```diff
@@ -0,0 +1,67 @@
+import SwiftUI
+
+struct ShippingLabelCarrierRow: View, Identifiable {
+
+    let id = UUID()
+    let title: String
+    let subtitle: String
+    let price: String
+    let image: UIImage?
+
+    var body: some View {
+        HStack(spacing: Constants.hStackSpacing) {
+            if let image = image {
+                ZStack {
+                    Image(uiImage: image).frame(width: Constants.imageSize, height: Constants.imageSize)
+                }
+                .frame(width: Constants.zStackWidth)
+            }
+            VStack(alignment: .leading,
+                   spacing: Constants.vStackSpacing) {
+                HStack {
+                    Text(title)
+                        .bodyStyle()
+                    Spacer()
+                    Text(price)
+                        .bodyStyle()
+                }
+                Text(subtitle)
+                    .footnoteStyle()
+            }
+        }
+        .padding([.top, .bottom], Constants.hStackPadding)
+        .padding([.leading, .trailing], Constants.padding)
+        .frame(minHeight: Constants.height)
+        .contentShape(Rectangle())
+    }
+}
+
+private extension ShippingLabelCarrierRow {
+    enum Constants {
+        static let zStackWidth: CGFloat = 48
+        static let vStackSpacing: CGFloat = 8
```

Comment:
There might be some opportunity to standardize the constants a bit in the future. Mainly following our standard Layout. P9MsLS-1g-p2

We probably would end up not needing some of the constants. 

Not sure if it's worth it though. 🤷 

```suggestion
    /// Global constants conforming to Woo Mobile layout standards
    /// (P9MsLS-1g-p2)
    enum Layout {
        /// Base length for spacing, gaps, padding.
        /// The name could be better XD
        static let baseLength: CGFloat = 8
    }

    enum Constants {
        /// We probably wouldn't even need these constants
        static let zStackWidth = Layout.baseLength * 6
        static let vStackSpacing = Layout.baseLength

```

Reply:
It would be great to have a standard for the Constants, but at the same time, just some spacing properties are always the same. For example, sometimes for the leading/trailing padding we use 8, or 16, or 20. We should also use a format that should be easily understandable, where we don't risk to broke anything modifying a constant in the future. It would be ideal for me to write a P2 post about this, to discuss this thing with the team. What do you think?

Reply:
Sounds good! Only if you have time though! I didn't want to add another thing to your plate. It's not a strong opinion for me at the moment since I'd need more experience with SwiftUI. 😄